src/Magnum/Math/StrictWeakOrdering.h

```diff
@@ -0,0 +1,93 @@
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::Math::StrictWeakOrdering
+ */
+
+namespace Magnum { namespace Math {
+/**
+@brief A functor which implements strict weak ordering for types in the math
+library.
+@param T the type for which to have strict weak ordering.
+
+A strict weak ordering enables a comparison operator between to elements of a
+set. With it an ordering can be achieved between elements.
+
+This is useful mainly for interoperability with ordering containers from the
+C++ standard library, like std::map and std::set. Instead of using this
+functor, it is recommended you use @ref StrictWeakOrderingFunc which provides
+a shorthand interface to this type.
+
+If you want to enable strict weak ordering through this or the
+@ref StrictWeakOrderingFunc shorthand, you must specialize this struct for your
+own type.
+ */
+template <class T> // Specialized for supported types
+struct StrictWeakOrdering {
+    /**
+     * @brief Compares two items
+     *
+     * Performs the comparison. For scalar types this is equivalent to
+     * operator <. For otther types specializations of this struct exist which
+     * have a specific implementation for the comparison such that is follows
+     * the definition of strict weak ordering.
+     */
+    bool operator()(const T& a, const T& b) const {
+        return a < b;
+    }
+};
+
+/**
+@brief A shorthand interface for @ref StrictWeakOrdering
+
+Allows the use of @ref StrictWeakOrdering in a shorter and more succinct way.
+To use for std::map or std::set, just add it as their compare template
+argument.
+
+@code{.cpp}
+std::set<Magnum::Vector2, Magnum::Math::StrictWeakOrderingFunc> mySet;
+std::map<Magnum::Vector4, int, Magnum::Math::StrictWeakOrderingFunc> myMap;
+@endcode
```

Comment:
I'm compiling all code snippets to ensure they don't get out of sync with future updates of the API. I'm not sure the Developers Guide mentions this, so:

- there's a file in `doc/snippets/MagnumMath.cpp` with a *huge* `main()` function inside
- add a scoped block somewhere to it and denoted by `/* [StrictWeakOrderingFunc] */` (the snippets are roughly alphabetically ordered), include `<map>` / `<set>` if needed and put the contents of the above `@code` inside
- reference the snippet using `@snippet MagnumMath.cpp StrictWeakOrderingFunc`
- the snippets are compile as part of `BUILD_TESTS`, so the CI might fail in case there's an error (or missing include etc.)

I can do the above myself post-merge if it's too much of a hassle for you.

Also, please omit the `Magnum::` prefix in all code snippets to avoid redundancy.